File: refactor_theme.dart
import 'dart:io';

void main() {
  final dir = Directory('d:/hand keyboard/lib');
  final files = dir.listSync(recursive: true).whereType<File>().where((f) => f.path.endsWith('.dart'));

  for (final file in files) {
    if (file.path.contains('app_theme.dart')) continue;

    var content = file.readAsStringSync();
    var original = content;

    content = content.replaceAll('AppTheme.surface0', 'Theme.of(context).scaffoldBackgroundColor');
    content = content.replaceAll('AppTheme.surface1', 'Theme.of(context).colorScheme.surface');
    content = content.replaceAll('AppTheme.surface2', 'Theme.of(context).colorScheme.surfaceContainerHighest');
    content = content.replaceAll('AppTheme.surface3', 'Theme.of(context).colorScheme.onSurface.withOpacity(0.1)');

    // Text colors and borders often use specific hardcoded hex values. 
    // We can replace some safe ones:
    content = content.replaceAll('const Color(0xFF1A1A1A)', 'Theme.of(context).colorScheme.surface');
    content = content.replaceAll('const Color(0xFF0D0D0D)', 'Theme.of(context).scaffoldBackgroundColor');
    content = content.replaceAll('const Color(0xFF252525)', 'Theme.of(context).colorScheme.surfaceContainerHighest');

    if (content != original) {
      file.writeAsStringSync(content);
      print('Updated ${file.path}');
    }
  }
}

File: refactor_theme.js
const fs = require('fs');
const path = require('path');

const walkSync = function(dir, filelist) {
  files = fs.readdirSync(dir);
  filelist = filelist || [];
  files.forEach(function(file) {
    if (fs.statSync(dir + '/' + file).isDirectory()) {
      filelist = walkSync(dir + '/' + file, filelist);
    }
    else {
      if (file.endsWith('.dart')) {
          filelist.push(path.join(dir, file));
      }
    }
  });
  return filelist;
};

const files = walkSync('d:/hand keyboard/lib');

for (const file of files) {
  if (file.includes('app_theme.dart')) continue;

  let content = fs.readFileSync(file, 'utf8');
  let original = content;

  content = content.replace(/AppTheme\.surface0/g, 'Theme.of(context).scaffoldBackgroundColor');
  content = content.replace(/AppTheme\.surface1/g, 'Theme.of(context).colorScheme.surface');
  content = content.replace(/AppTheme\.surface2/g, 'Theme.of(context).colorScheme.surfaceContainerHighest');
  content = content.replace(/AppTheme\.surface3/g, 'Theme.of(context).colorScheme.onSurface.withOpacity(0.1)');
  
  if (content !== original) {
    fs.writeFileSync(file, content, 'utf8');
    console.log(`Updated ${file}`);
  }
}
